### ISSUE TITLE
1000 ideas

### ISSUE THREAD

**Issue by geekley** [icebound777/PMR-SeedGenerator#45]
First of all, this project is really really cool! I'm having way too much fun with it!

I don't think any of these are important, but I'd like to throw some ideas here, just for funzies.
The version I'm playing now is 0.13.2 btw, so sorry if something is outdated here.

### Difficulty: Option to make more things cost money
To avoid stashing money and items and to make battles more meaningful, it would be nice to have more opportunities to spend money.
#### Sleeping
With this option on, Toad Houses would become inns, like in TTYD. Same applies to Mario's bed: you have to leave money to help Luigi pay the bills. The cost would be the same everywhere, but increase proportionately to how much it can help you.
For example: `Price = MaxHP/5 + MaxFP/5 + 1*MaxSP`
#### Recovery Blocks
A related option would also affect Heart Blocks. You could add the following options besides Vanilla and No blocks:
a. they can be used only once
b. they cost coins to use like in TTYD - they would increase cost similar to above, but cheaper, e.g. `MaxHP/5 + MaxFP/5`
#### Home warping
Warping home could also cost some part of your coins, e.g 50% (you can still always use it, even at zero coins).
#### Cooking
Maybe allow setting a cost for cooking as well? It's a bit weird to be able to upgrade items for free.

### Minor improvement: Reorder key items
So that the ones you don't need to worry about after collecting them (you keep them forever, and don't use actively, e.g. Ultra Stone, Lucky Star, Silver/Gold Credit, etc) are placed at the end. A usable item like home warping should be first.

### Minor improvement: Show number of items on C-left menu
So you can check if you're carrying too many items without having to pause or count. E.g. instead of "Items", menu would say "N item(s)"  or maybe "N/T items" where N is how many you're carrying, T is capacity.

### Cosmetics/Mario: "Unlock when leveling up" option
This would make Mario skins a random sequence defined by the seed. He starts with first color. As he levels up, he occasionally unlocks a new color until he goes through all colors, ending with the last color only at Max Level.
Maybe you could also allow switching colors at the pause menu, pressing (A) in Mario's Level explanations.

### Shopsanity: Fill empty shop slots with non-essential items
This would make it so after buying one-time items in shops, slot won't be empty, but instead have a random non-essential-but-maybe-useful item in its place. You can make it really expensive.

### Major feature: "Cooksanity"
It would be cool to have a mode where you have to cook all (or a specified amount of) recipes to be able to beat the game. Then, when cooking the required amount of recipes, Tayce T. would give some key item (e.g. a Chef's Diploma) that would be necessary somehow to reach the final boss. To help with that, you could add a recipe tracker like the current tracker for key items. Replenishable base ingredients for the recipes would be in logic, of course. This would be interesting with having to pay a cooking price as well.

---
Lastly (not complaining or anything), just a comment on few things I found weird/funny...
- Getting Gold Credit first than silver opens both games, making Silver Credit useless.
- Merlow can sell a Star Piece... and charge multiple star pieces for it.
- Home warping item could have been a necklace or something (Lucky Star?), but instead it's a mushroom that can be eaten forever... it's a miracle! The multiplication of the mushrooms :laughing: I guess that's why Mario never gets hungry!
